### PR TITLE
[prometheus] metric for upstream status / proxy status

### PR DESCRIPTION
**Pull request by mikz**
Metric `upstream_status` with label `status` is incremented with HTTP status received from upstream.
Metric `cors_proxy_status` with label `status` is incremented when there is no upstream and CORS proxy terminates the request (403, 503, ...)

Example:
```
# HELP cors_proxy_database_connection Database Connection State
# TYPE cors_proxy_database_connection gauge
cors_proxy_database_connection{state="state"} 1
# HELP cors_proxy_status HTTP status generated by CORS proxy
# TYPE cors_proxy_status counter
cors_proxy_status{status="403"} 1
cors_proxy_status{status="503"} 1
# HELP nginx_http_connections Number of HTTP connections
# TYPE nginx_http_connections gauge
nginx_http_connections{state="accepted"} 0
nginx_http_connections{state="active"} 2
nginx_http_connections{state="handled"} 0
nginx_http_connections{state="reading"} 0
nginx_http_connections{state="total"} 0
nginx_http_connections{state="waiting"} 1
nginx_http_connections{state="writing"} 1
# HELP nginx_metric_errors_total Number of nginx-lua-prometheus errors
# TYPE nginx_metric_errors_total counter
nginx_metric_errors_total 0
# HELP upstream_status HTTP status from upstream servers
# TYPE upstream_status counter
upstream_status{status="201"} 1
```

### CODE REVIEW

File: src/cors-proxy/init.lua [hunk 1]
```diff
@@ -84,6 +86,14 @@ local function cors_preflight()
 end
 
 function _M:log()
+  local upstream_status = tonumber(ngx.var.upstream_status)
```
Comment (mikz) [inline src/cors-proxy/init.lua:89]: `upstream_status` can be `-` when balancer phase exits before setting the peer.

**Review by maneta — APPROVED** [review]
👍

**Comment by maneta** [comment]
This should close #5 right?

**Comment by mikz** [comment]
I guess. Buy you also mention there response time, which this PR does not do.

**Comment by maneta** [comment]
@mikz yes but I think that maybe implementing response time and such would generate high cardinality, which would increase the amount of data stored in Prometheus, as that every unique combination of key-value label pairs represents a new time series.

What do you think?

**Comment by mikz** [comment]
@maneta well, response times should obviously should not be stored as raw values. 

As described in https://prometheus.io/docs/practices/histograms/

**Comment by mikz** [comment]
@maneta merging this. It implements only part of https://github.com/3scale/cors-proxy/issues/5. If you  want response times then propose the format.

**Pull request merged**